lib/db_utils.go

```diff
@@ -573,12 +573,13 @@ type DBPrefixes struct {
 	// Note, we parse the ValidatorPKID and StakerPKID from the key.
 	PrefixSnapshotStakeToRewardByValidatorAndStaker []byte `prefix_id:"[92]" is_state:"true"`
 
-	// PrefixLockedBalanceEntryByHODLerPKIDProfilePKIDUnlockTimestampNanoSecs:
+	// PrefixLockedBalanceEntry:
 	// Retrieves LockedBalanceEntries that may or may not be claimable for unlock.
-	// LockedBalanceEntries can be retrieved by HodlerPKID and CreatorPKID are have their
-	// corresponding unlock timestamp appended to sort by timestamp.
-	// Prefix, <HodlerPKID [33]byte>, <ProfilePKID [33]byte>, <UnlockTimestampNanoSecs int64> -> <LockedBalanceEntry>
-	PrefixLockedBalanceEntryByHODLerPKIDProfilePKIDUnlockTimestampNanoSecs []byte `prefix_id:"[93]" is_state:"true"`
+	// A discriminator byte it placed before the UnlockTimestampNanoSecs to allow for separation
+	// among the vested and unvested locked balance entries without separate indexes.
+	// Prefix, <HodlerPKID [33]byte>, <ProfilePKID [33]byte>, <vested/unvested byte>,
+	// <UnlockTimestampNanoSecs [8]byte>, <VestingEndTimestampNanoSecs [8]byte> -> <LockedBalanceEntry>
+	PrefixLockedBalanceEntry []byte `prefix_id:"[93]" is_state:"true"`
```

Comment:
Kinda feels like you don't need to add the vested/unvested byte, since it kinda makes you go through two lookups for no reason (one for vested and then another for unvested). Alternatively, you could just look up everything that's unlockable and then split them by UnlockTimestamp != VestingEndTime or whatever. But there also doesn't seem to be any harm in it so I'd keep it this way if you like it.